posthog/hogql_queries/legacy_compatibility/filter_to_query.py

```diff
@@ -539,6 +539,7 @@ def _insight_filter(filter: dict, allow_variables: bool = False):
                 ),
                 period=filter.get("period"),
                 showMean=filter.get("show_mean"),
+                meanRetentionCalculation=filter.get("mean_retention_calculation"),
```

Comment:
nit: the conversion in this method should do the same thing that we do frontend side i.e. in frontend/src/queries/nodes/InsightQuery/utils/filtersToQueryNode.ts. here you had this code:

```ts
 meanRetentionCalculation: filters.mean_retention_calculation || (typeof filters.show_mean === 'boolean' ? (filters.show_mean ? 'simple' : 'none') : 'simple'),
```

it doesn't really matter in this case, as there shouldn't be legacy insights with a show_mean filter. just commenting this for completeness.